src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,55 +1,104 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
 import './FindMovie.scss';
 
 import { MovieCard } from '../MovieCard';
-import movies from '../../api/movies.json';
-
-export const FindMovie: React.FC = () => (
-  <>
-    <form className="find-movie">
-      <div className="field">
-        <label className="label" htmlFor="movie-title">
-          Movie title
-        </label>
-
-        <div className="control">
-          <input
-            type="text"
-            id="movie-title"
-            placeholder="Enter a title to search"
-            className="input is-danger"
-          />
-        </div>
+import { request } from '../../api/api';
 
-        <p className="help is-danger">
-          Can&apos;t find a movie with such a title
-        </p>
-      </div>
+interface Props {
+  onAdd: (movie: Movie | null) => void;
+}
+
+export const FindMovie: React.FC<Props> = (props) => {
+  const { onAdd } = props;
+  const [newMovie, setNewMovie] = useState(null as Movie | null);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [isFound, setIsFound] = useState(true);
+
+  const getMovieFromServer = async () => {
+    const data = await request(searchQuery);
+
+    if (data.Response === 'False') {
+      setIsFound(false);
+      setNewMovie(null);
+
+      return;
+    }
 
-      <div className="field is-grouped">
-        <div className="control">
-          <button
-            type="button"
-            className="button is-light"
-          >
-            Find a movie
-          </button>
+    setIsFound(true);
+    setNewMovie({
+      title: data.Title,
+      description: data.Plot,
+      imgUrl: data.Poster,
+      imdbId: data.imdbID,
+      imdbUrl: `https://www.imdb.com/title/${data.imdbID}`,
+    });
+    setSearchQuery('');
+  };
```

Comment:
That was the point I was wodering the most: where should we place the logic in FC.
Nice one!